pkg/vpcmodel/semanticDiff.go

```diff
@@ -271,6 +246,59 @@ func (connDiff *connectivityDiff) string(diffAnalysis diffAnalysisType, thisMinu
 	return res
 }
 
+// connDiffEncode encodes connectivesDiff information for grouping:
+// this includes the following 4 strings separated by ";"
+//  1. diff-type info: e.g. diff-type: removed
+//  2. connection of config1
+//  3. connection of config2
+//  4. info regarding missing endpoints: e.g. vsi0 removed
+func connDiffEncode(src, dst VPCResourceIntf, connDiff *connectionDiff,
+	thisMinusOther bool) string {
+	conn1Str, conn2Str := conn1And2Str(connDiff, thisMinusOther)
+	diffType, endpointsDiff := diffAndEndpointsDisc(connDiff.diff, src, dst, thisMinusOther)
+	return diffType + semicolon + conn1Str + semicolon + conn2Str + semicolon + endpointsDiff
+}
+
+func diffInfoStr(diffAnalysis diffAnalysisType) string {
+	if diffAnalysis == Subnets {
+		return "subnets-diff-info:"
+	} else if diffAnalysis == Vsis {
+		return "vsis-diff-info:"
+	}
+	return ""
+}
+
+func conn1And2Str(connDiff *connectionDiff, thisMinusOther bool) (conn1Str, conn2Str string) {
+	if thisMinusOther {
+		conn1Str = connStr(connDiff.conn1)
+		conn2Str = connStr(connDiff.conn2)
+	} else {
+		conn1Str = connStr(connDiff.conn2)
+		conn2Str = connStr(connDiff.conn1)
+	}
+	return conn1Str, conn2Str
+}
+
+// printDiffLine print one diff line, using the above string as input
+func printDiffLine(diffAnalysis diffAnalysisType, src, dst EndpointElem, decoded string) string {
+	decodedDetails := strings.Split(decoded, semicolon)
```

Comment:
Instead of encoding/decoding strings, can you extend the type `GroupedConnLine` to contain the relevant objects?

Reply:
I think the current solution is better.
The alternative is to extend `GroupedConnLine`  as you suggest and to change the actual grouping, adding the new objects to the indexing when the grouping is done for "diffs". Resulting in more code which will not be easier to maintain. 
Grouping is done after all the relevant computation, and we need these objects only for grouping indexes and for printing. Which is why storing it as string suffices for our needs and simplifies the code and the `GroupedConnLine`  data structure.

Reply:
why extending `GroupedConnLine` will change the actual grouping? other fields can be added without changing the indexing or the current grouping mechanism.

Reply:
`addPublicConnectivity` is based on the current encoding. Do you suggest changing that?
Do you want to take it offline?

Reply:
done